src/plugins/part.c

```diff
@@ -2006,6 +2013,119 @@ gboolean bd_part_set_part_id (const gchar *disk, const gchar *part, const gchar
     return TRUE;
 }
 
+/**
+ * bd_part_set_part_uuid:
+ * @disk: device the partition belongs to
+ * @part: partition the UUID should be set for
+ * @uuid: partition UUID to set
```

Comment:
So does it mean that when `uuid == NULL`, it means the UUID value will be cleared from the partition table? Another option is to generate a new random value (we might do that in UDisks).

Reply:
I don't think it's possible to remove the UUID. The function now doesn't accept `NULL`, none of the `set_X` functions in the part plugin take `NULL`, if you want to remove the value (for things that can be removed, like name), you need to use an empty string for that. We can change it to generate a new random one, that would be similar to how the `bd_fs_*_set_uuid` functions work for filesystem.

Reply:
Hmm, each of the `bd_fs_*_set_uuid` functions take slightly different "magic" values according to the `mkfs` tool used. And the `part` plugin setters only seems to just pass the `NULL` value further down the `libfdisk` code to see what happens. Okay, let's keep it as proposed, at least it's consistent with the others.